shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java

```diff
@@ -61,20 +60,88 @@
           .argName("replicas")
           .desc("Number of block replicas of each loaded file, default: " + DEFAULT_REPLICATION)
           .build();
-  private static final int DEFAULT_PARALLELISM = 10;
-  private static final Option PARALLELISM_OPTION =
+
+  private static final int DEFAULT_ACTIVE_JOBS = 1000;
+  private static final Option ACTIVE_JOBS_OPTION =
       Option.builder()
-          .longOpt("parallelism")
+          .longOpt("active_jobs")
           .required(false)
           .hasArg(true)
-          .argName("# concurrent operations")
           .numberOfArgs(1)
-          .desc("Number of concurrent load operations, default: " + DEFAULT_PARALLELISM)
+          .type(Number.class)
+          .argName("jobs")
+          .desc("Maximum number of active outgoing jobs, default: " + DEFAULT_ACTIVE_JOBS)
           .build();
 
-  private ThreadPoolExecutor mDistributedLoadServicePool;
-  private CompletionService<AlluxioURI> mDistributedLoadService;
-  private ArrayList<Future<AlluxioURI>> mFutures = new ArrayList<>();
+  private class JobAttempt {
+    private final JobConfig mJobConfig;
+    private final RetryPolicy mRetryPolicy;
+    private final JobMasterClient mClient;
+
+    private Long mJobId;
+
+    private JobAttempt(JobConfig jobConfig, RetryPolicy retryPolicy, ClientContext clientContext) {
+      mJobConfig = jobConfig;
+      mRetryPolicy = retryPolicy;
+      mClient = JobMasterClient.Factory.create(
+          JobMasterClientContext.newBuilder(clientContext).build());
+    }
+
+    private boolean run() {
+      if (mRetryPolicy.attempt()) {
+        mJobId = null;
+        try {
+          mJobId = mClient.run(mJobConfig);
+        } catch (IOException e) {
+          LOG.warn("Failed to get status for job (jobId={})", mJobId, e);
+          // Do nothing. This will be counted as a failed attempt
+        }
+        return true;
+      }
+      return false;
+    }
+
+    /**
+     * Returns the status of the job attempt.
+     * @return True if finished successfully or cancelled, False if FAILED and should be retried,
+     *              null if the status should be checked again later
+     */
+    private Boolean check() {
```

Comment:
Instead of returning a boolean should we just use the state, I'm not sure we are getting that much simplification?